### PR TITLE
Fix full LVM encrypt on non-UEFI architectures

### PR DESCRIPTION
**Pull request by dzedro**
https://progress.opensuse.org/issues/15926

[x86_64](http://10.100.12.155/tests/5068)
[s390x](http://10.100.12.155/tests/5067)
[ppc64le](https://openqa.suse.de/tests/784721)

### CODE REVIEW

File: products/sle/main.pm [hunk 1]
```diff
@@ -570,7 +570,7 @@ sub load_reboot_tests() {
                 loadtest "installation/boot_into_snapshot";
             }
         }
-        if (get_var('ENCRYPT')) {
+        if (get_var('ENCRYPT') && !check_var('BACKEND', 'svirt')) {
```
Comment (okurz) [inline products/sle/main.pm]: It's not obvious why svirt should be excluded. Can you comment? Or even better extract a "is…applicable" function

File: tests/installation/reconnect_s390.pm [hunk 2]
```diff
@@ -40,6 +40,20 @@ sub run() {
...
+            $svirt->suspend;
+            type_string "export pty=`virsh dumpxml $name | grep \"console type=\" | sed \"s/'/ /g\" | awk '{ print \$5 }'`\n";
+            type_string "echo \$pty\n";
+            $svirt->resume;
+
+            wait_serial("Please enter passphrase for disk.*", 100);
+            type_string "echo $password > \$pty\n";
+            wait_serial("Please enter passphrase for disk.*", 100);
+            type_string "echo $password > \$pty\n";
+        }
```
Comment (okurz) [inline tests/installation/reconnect_s390.pm]: shouldn't that better be moved into the encrypted password typing module? typing password for encrypt is not really something that we would expect in "reconnect_s390", is it?

Comment (Soulofdestiny) [inline tests/installation/reconnect_s390.pm]: well... the reconnect_s390 covers everything related to rebooting the machine
the password needs to be typed twice during the (re)boot process... 
We have now to options:
1. move this to the encrypted password module, but also move (and duplicate) the wait for the system being booted 
2. leave it like this and live with it

**Comment by dzedro** [comment]
It can be like now ... 
http://10.100.12.155/tests/5081

File: lib/utils.pm [hunk 3]
```diff
@@ -59,9 +59,25 @@ sub unlock_if_encrypted {
...
+        my $name     = $svirt->name;
+        $svirt->suspend;
+        type_string "export pty=`virsh dumpxml $name | grep \"console type=\" | sed \"s/'/ /g\" | awk '{ print \$5 }'`\n";
+        type_string "echo \$pty\n";
+        $svirt->resume;
+
+        wait_serial("Please enter passphrase for disk.*", 100);
+        type_string "echo $password > \$pty\n";
+        wait_serial("Please enter passphrase for disk.*", 100);
+        type_string "echo $password > \$pty\n";
```
Comment (okurz) [inline lib/utils.pm]: why is it necessary twice?

**Comment by okurz** [comment]
LGTM

File: lib/utils.pm (hunk 3 above)
Comment (dzedro) [inline lib/utils.pm]: @Soulofdestiny bug or s390x is just super secure ?

Comment (Soulofdestiny) [inline lib/utils.pm]: my guess would be the extra /boot/zipl partition which is also encrypted
the first passphrase prompt is before grub and the second is after grub

Comment (dzedro) [inline lib/utils.pm]: @Soulofdestiny like this ?

File: products/sle/main.pm [hunk 4]
```diff
@@ -572,6 +573,9 @@ sub load_reboot_tests() {
         }
         if (get_var('ENCRYPT')) {
             loadtest "installation/boot_encrypt";
+            if (check_var('BACKEND', 'svirt')) {
+                loadtest "installation/reconnect_s390";
+            }
```
Comment (dzedro) [inline products/sle/main.pm]: I also added reconnect behind boot_encrypt to make it pass.

[failed](http://10.100.12.155/tests/5082)
[passed](http://10.100.12.155/tests/5086)

Comment (dzedro) [inline products/sle/main.pm]: ok have to finish this somehow to make it pass

Comment (Soulofdestiny) [inline products/sle/main.pm]: you should also do a check_var('ARCH', 's390x') to prevent it of being executed on other svirt jobs

**Comment by okurz** [comment]
> my guess would be the extra /boot/zipl partition which is also encrypted
> the first passphrase prompt is before grub and the second is after grub

good, add a comment

**Comment by dzedro** [comment]
Is OK I guess
http://10.100.12.155/tests

**Pull request merged**